src/logger.rs

```diff
@@ -0,0 +1,87 @@
+use ansi_term::Color;
+use log::{Level, LevelFilter, Metadata, Record};
+use std::{
+    env,
+    fs::{self, File, OpenOptions},
+    io::Write,
+    sync::{Arc, Mutex},
+};
+
+pub struct StarshipLogger {
+    log_file: Arc<Mutex<File>>,
+    log_file_content: Arc<String>,
```

Comment:
Should `log_file` be `BufWrite` and `log_file_content` a `HashSet<String>`? I think compressing the logfile could also be worth it.

Reply:
I've changed them to your respective suggestions but how do you think we should compress the log files.
Because we can't just compress all logs that are not our session (e.g. from an old session) because we can have multiple sessions open at the same time.
And compressing our own file does not make much sense to me because we would have to uncompress and recompress the log file every starship invocation which could potentially take quite long.

Reply:
It looks like the file gets flushed with every log message, so no Bufwriter might be better because it's not going to buffer anything.

I didn't think of having to recompress every time, I guess it's better not to compress that in that case.